server/app/controllers/admin/AdminProgramBlockPredicatesController.java

```diff
@@ -150,6 +180,78 @@ public Result update(Request request, long programId, long blockDefinitionId) {
+
+    if (predicateFormWrapper.hasErrors()) {
+      StringBuilder errorMessageBuilder = new StringBuilder("Did not save eligibility condition:");
+      predicateFormWrapper
+          .errors()
+          .forEach(error -> errorMessageBuilder.append(String.format("\n• %s", error.message())));
+
+      return redirect(
+              routes.AdminProgramBlockPredicatesController.editEligibility(
+                  programId, blockDefinitionId))
+          .flashing("error", errorMessageBuilder.toString());
+    }
+
+    // TODO(https://github.com/seattle-uat/civiform/issues/322): Implement complex predicates.
+    //  Right now we only support "leaf node" predicates (a single logical statement based on one
+    //  question). In the future we should support logical statements that combine multiple "leaf
+    //  node" predicates with ANDs and ORs.
+    BlockVisibilityPredicateForm predicateForm = predicateFormWrapper.get();
+
+    Scalar scalar = Scalar.valueOf(predicateForm.getScalar());
+    Operator operator = Operator.valueOf(predicateForm.getOperator());
+    PredicateValue predicateValue =
+        parsePredicateValue(
+            scalar,
+            operator,
+            predicateForm.getPredicateValue(),
+            predicateForm.getPredicateValues());
+
+    LeafOperationExpressionNode leafExpression =
+        LeafOperationExpressionNode.create(
+            predicateForm.getQuestionId(), scalar, operator, predicateValue);
+    PredicateAction action = PredicateAction.valueOf(predicateForm.getPredicateAction());
+    PredicateDefinition predicateDefinition =
+        PredicateDefinition.create(PredicateExpressionNode.create(leafExpression), action);
+
+    try {
+      programService.setBlockEligibilityDefinition(
+          programId,
+          blockDefinitionId,
+          Optional.of(EligibilityDefinition.builder().setPredicate(predicateDefinition).build()));
+    } catch (ProgramNotFoundException e) {
+      return notFound(String.format("Program ID %d not found.", programId));
+    } catch (ProgramBlockDefinitionNotFoundException e) {
+      return notFound(
+          String.format("Block ID %d not found for Program %d", blockDefinitionId, programId));
+    } catch (IllegalPredicateOrderingException e) {
+      return redirect(
+              routes.AdminProgramBlockPredicatesController.edit(programId, blockDefinitionId))
+          .flashing("error", e.getLocalizedMessage());
+    }
+
+    ReadOnlyQuestionService roQuestionService =
+        questionService.getReadOnlyQuestionService().toCompletableFuture().join();
+
+    return redirect(
+            routes.AdminProgramBlockPredicatesController.editEligibility(
+                programId, blockDefinitionId))
+        .flashing(
+            "success",
+            String.format(
+                "Saved eligibility condition: %s %s",
+                action.toDisplayString(),
+                leafExpression.toDisplayString(roQuestionService.getUpToDateQuestions())));
+  }
```

Comment:
note (non-blocking): this introduces quite a bit of code duplication with the visibility versions of these methods. Since I'm planning to rewrite a bunch of the logic here to support multi-question predicates soon I think it's ok to merge it as-is.